src/ncp/posix/netif.cpp

```diff
@@ -453,12 +453,30 @@ void Netif::ProcessMldEvent(void)
             case kIcmpv6Mldv2ModeIsExcludeType:
                 error = OTBR_ERROR_NONE;
                 break;
+            ///< Only update subscription on NCP when the target multicast address is not in `mIp6MulticastAddresses`.
+            ///< This indicates that this is the first time the multicast address subscription needs to be updated.
             case kIcmpv6Mldv2RecordChangeToIncludeType:
-                error = (record->mNumSources == 0) ? mDeps.Ip6MulAddrUpdateSubscription(address, /* isAdd */ false)
-                                                   : OTBR_ERROR_NONE;
+                if (std::find(mIp6MulticastAddresses.begin(), mIp6MulticastAddresses.end(), Ip6Address(address)) !=
+                    mIp6MulticastAddresses.end())
+                {
+                    error = (record->mNumSources == 0) ? mDeps.Ip6MulAddrUpdateSubscription(address, /* isAdd */ false)
```

Comment:
should the mIp6MulticastAddresses be updated if this succeeds?

Reply:
After the host send spinel command to NCP and NCP successfully updates the multicast address, NCP will sync the address table to the host. At that time, we will update `mIp6MulticastAddresses`. So at this moment, the multicast addresses on NCP hasn't been updated. We should update `mIp6MulticastAddresses` after NCP has successfully updated the multicast address. In this way, we use single source of truth to maintain the addresses. `mIp6MulticastAddresses` will always follow the IP6 multicast addresses on NCP.